src/layouts/EditNavBar.jsx

```diff
@@ -672,43 +633,215 @@ const EditNavBar = ({ match }) => {
+                                        index={linkIndex}
+                                        errors={errors.links[linkIndex]}
+                                      />
+                                    </Editable.DraggableAccordionItem>
+                                  )}
+                                  {link.collection && (
+                                    <Editable.DraggableAccordionItem
+                                      index={linkIndex}
+                                      tag={<Tag variant="subtle">Folder</Tag>}
+                                      title={link.title}
+                                      isInvalid={_.some(
+                                        errors.links[linkIndex]
+                                      )}
+                                    >
+                                      <FolderMenuBody
+                                        {...link}
+                                        index={linkIndex}
+                                        errors={errors.links[linkIndex]}
+                                        options={options}
+                                      />
+                                    </Editable.DraggableAccordionItem>
+                                  )}
+                                  {link.sublinks && (
+                                    <Editable.DraggableAccordionItem
+                                      index={linkIndex}
+                                      tag={
+                                        <Tag variant="subtle">Menu Group</Tag>
+                                      }
+                                      title={link.title}
+                                      isInvalid={
+                                        _.some(errors.links[linkIndex]) ||
+                                        _.some(
+                                          errors.sublinks[
+                                            linkIndex
+                                          ].map((sublink) => _.some(sublink))
+                                        )
+                                      }
+                                    >
+                                      <GroupMenuBody
+                                        {...link}
+                                        index={linkIndex}
+                                        errors={{
+                                          ...errors.links[linkIndex],
+                                          sublinks: errors.sublinks[linkIndex],
+                                        }}
+                                      />
+                                    </Editable.DraggableAccordionItem>
+                                  )}
+                                  {!link.resource_room &&
+                                    !link.collection &&
+                                    !link.sublinks && (
+                                      <Editable.DraggableAccordionItem
```

Comment:
could you point me to the corresponding section the legacy code that has this condition? just wanted to ensure we aren't causing a regression

Reply:
It's not exactly as it was before, the closest section to this is `NavElem` in `NavSection`

In the past, the different types of navigation elements actually were all the same type of component, with fields being conditionally rendered based on the presence/absence of params like `resource_room` and `collection`

Specifically, in the past, to render a url field the equivalent code is
```
{!collection && !isResource && (
  <FormContext hasError={!!linkErrors.url}>
    <FormTitle>Permalink</FormTitle>
    <FormField
      placeholder="Permalink"
      id={`link-${linkIndex}-url`}
      value={url}
      onChange={onFieldChange}
    />
    <FormError>{linkErrors.url}</FormError>
  </FormContext>
)}
```

But this was also because the `submenu` type of menu item also contains the `url` param. Since these items are all separate components now, in order to definitively identify this component as a `single page` type of menu, we can only check this via the absence of the distinguishing features of the other 3 types

Reply:
my main fear is that this might lead to issues down the line because the conditionals don't match exactly. things we'd need to check here are:
1. is `isResource` actually a `boolean` or a falsy value? because `!{}` evaluates to `false`, if `isResource` is a `boolean`, we might have a regression (if `resource_room` is `{}`)
2. there's an additional check for `sublinks` which wasn't present before - not sure if this will cause any issues.

Reply:
1. `isResource` corresponds to `link.resource_room` previously, and we're checking for `!link.resource_room` now! The actual check in our template code is assuming `resource_room` is a boolean value (in fact, the only value it can take is `true`)
2. `sublinks` is to catch the menu group type items - previously the field was rendered in both menu group and single page, so we didn't have a check for that.

I think what we can verify here is that

- Three of the sections have a unique identifying detail (`Group` - `sublinks`, `Resource room` - `resource_room`, `Folder` - `collection`), but `Group` is a superset of `Single page`, so it has no unique identifying feature - it thus becomes the fallback component
- These details are also what we're searching for in our template, so it should be okay

Reply:
sgtm, merge whenever